### PR TITLE
Remove NGrok. Configure Tunnel URL

### PR DESCRIPTION
**Pull request by rido-min**
Removes NGrok dependencies.

To test remote bots, the user has to configure a tunnel - any tunnel software will work - pointing to the emulator service. The tunnel needs to be configured to target to the actual port used by the emulator service.

The port is shown in the settings page and is also available to copy to the clipboard via File Menu.

I've tested with Ngrok, and [Microsoft.devTunnel](https://learn.microsoft.com/en-us/azure/developer/dev-tunnels/)

Version bumped to `4.15.1` availabe to download for testing from ADO: [BotFramework-Emulator-4.15.1-windows-setup.exe](https://artprodeus21.artifacts.visualstudio.com/Aae8b0170-8c49-492d-afe5-8666d1955c7e/18091c08-0c2a-44b2-9078-2243ba6be722/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL0Z1c2VMYWJzL3Byb2plY3RJZC8xODA5MWMwOC0wYzJhLTQ0YjItOTA3OC0yMjQzYmE2YmU3MjIvYnVpbGRJZC8zOTUzOTEvYXJ0aWZhY3ROYW1lL2Ryb3BfYnVpbGRfd2luZG93cw2/content?format=file&subPath=%2FBotFramework-Emulator-4.15.1-windows-setup.exe)

### CODE REVIEW

File: packages/app/client/src/ui/editor/appSettingsEditor/appSettingsEditor.tsx [hunk 1]
```diff
@@ -88,33 +82,28 @@ export class AppSettingsEditor extends React.Component<AppSettingsEditorProps, A
...
-      localhost = '',
-      locale = '',
       use10Tokens = false,
       useCodeValidation = false,
       userGUID = '',
       autoUpdate = false,
       usePrereleases = false,
       collectUsageData = false,
+      tunnelUrl = '',
+      localPort = 321,
```
Comment (rido-min) [inline packages/app/client/src/ui/editor/appSettingsEditor/appSettingsEditor.tsx:106]: change to 0

File: packages/app/client/src/ui/editor/appSettingsEditor/appSettingsEditor.tsx [hunk 2]
```diff
@@ -361,7 +268,9 @@ export class AppSettingsEditor extends React.Component<AppSettingsEditorProps, A
     this.props.setDirtyFlag(dirty);
   }
 
-  private setNgrokInputRef = (ref: HTMLInputElement): void => {
-    this.pathToNgrokInputRef = ref;
+  private getLocalPort = async () => {
+    const lp = await ipcRenderer.invoke('local-server-port');
+    console.log('LOCAL PORT', lp);
```
Comment (OEvgeny) [inline packages/app/client/src/ui/editor/appSettingsEditor/appSettingsEditor.tsx:273]:
```suggestion
```

File: tsconfig.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
     "noImplicitThis": false,
     "noImplicitAny": false,
     "allowSyntheticDefaultImports": true,
-    "suppressImplicitAnyIndexErrors": true,
+    // "suppressImplicitAnyIndexErrors": true,
```
Comment (OEvgeny) [inline tsconfig.json:18]:
```suggestion
```

File: packages/app/client/src/ui/editor/appSettingsEditor/appSettingsEditor.tsx (hunk 1 above)
Comment (OEvgeny) [reply packages/app/client/src/ui/editor/appSettingsEditor/appSettingsEditor.tsx:106]: @rido-min should we change to 0 in scope of the PR?

**Comment by OEvgeny** [comment]
Looks good. I think the only piece which is missing right now is UI changes for ngrok tab. See `packages/app/client/src/ui/shell/mdi/tab/ngrokTab.tsx`.

As we are probably not able to track status of the tunnel, we may remove the tab.

File: packages/app/client/src/ui/editor/appSettingsEditor/appSettingsEditor.tsx (hunk 1 above)
Comment (rido-min) [reply packages/app/client/src/ui/editor/appSettingsEditor/appSettingsEditor.tsx]: yep, changed

**Comment by rido-min** [comment]
> As we are probably not able to track status of the tunnel, we may remove the tab.

yep, removed in last commit

**Review by ceciliaavila — COMMENTED** [review]
I can see several references to ngrok in the specs. Also, a few dedicated ngrok tests. Should we keep these? 
e.g.:
[logEntry.spec.tsx](https://github.com/microsoft/BotFramework-Emulator/blob/4e00746918fef4b90bde85d80e7546fa273e41c3/packages/app/client/src/ui/editor/emulator/parts/log/logEntry.spec.tsx#L319)
[util.spec.ts](https://github.com/microsoft/BotFramework-Emulator/blob/71fe728f278d900e33fe2227abed466279647343/packages/sdk/shared/src/types/log/util.spec.ts#L130)

**Comment by rido-min** [comment]
> I can see several references to ngrok in the specs. Also, a few dedicated ngrok tests. Should we keep these? e.g.: [logEntry.spec.tsx](https://github.com/microsoft/BotFramework-Emulator/blob/4e00746918fef4b90bde85d80e7546fa273e41c3/packages/app/client/src/ui/editor/emulator/parts/log/logEntry.spec.tsx#L319) [util.spec.ts](https://github.com/microsoft/BotFramework-Emulator/blob/71fe728f278d900e33fe2227abed466279647343/packages/sdk/shared/src/types/log/util.spec.ts#L130)

I was not looking at test runs, I can remove all references to ngrok in tests as well.

**Comment by rido-min** [comment]
@ceciliaavila spec.ts files updated to remove any reference to ngrok in https://github.com/microsoft/BotFramework-Emulator/pull/2461/commits/45635ca6b823350b275ece274dda3fcd7e55f93a

**Pull request merged**